src/librustc_typeck/check/method/suggest.rs

```diff
@@ -173,7 +173,11 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                      if mode == Mode::MethodCall {
                                          "method"
                                      } else {
-                                         "associated item"
+                                         if item_name.as_str().chars().next().unwrap().is_lowercase() {
```

Comment:
Could you avoid `unwrap`? It may break during error recovery, if the item name is dummy.
(I haven't looked if this error recovery is actually performed for associated items right now, but anyway.)